pennylane/beta/plugins/default_tensor.py

```diff
@@ -355,41 +366,74 @@ class DefaultTensor(Device):
     C_DTYPE = np.complex128
     R_DTYPE = np.float64
 
-    def __init__(self, wires, shots=1000, analytic=True):
+    zero_state = np.array([1.0, 0.0], dtype=C_DTYPE)
+
+    def __init__(self, wires, shots=1000, analytic=True, representation="exact"):
         super().__init__(wires, shots)
-        self.analytic = True
-        self._nodes = []
-        self._edges = []
-        self._state_node = None
-        self._free_edges = []
+        self.analytic = analytic
+        self._rep = representation
         self.reset()
 
-    @staticmethod
-    def _create_basis_state(state, wires):
-        """Helper function to create a basis state with the correct shape.
+    def reset(self):
+        """Reset the device"""
+        self._clear_network()
 
-        Args:
-            state (array[int]): array of 0s and 1s of size ``(wires,)`` representing
-                the basis state
-            wires (list[int]): the wires the basis state should
-                be prepared on
+        # prepare a factorized all-zeros state
+        self._add_initial_state_nodes(
+            [self.zero_state] * self.num_wires,
+            [[w] for w in range(self.num_wires)],
+            ["ZeroState"] * self.num_wires,
+        )
 
-        Returns:
-            array[int]: state array of size ``[2]*len(wires)``
+    def _clear_network(self):
+        """Remove all data representing the current network from internal cache."""
+        self._nodes = {}
+        if self._rep == "exact":
+            self._contracted = False
+            self._terminal_edges = []
+
+        elif self._rep == "mps":
+            raise NotImplementedError
+            # nodes = []
+            # for w in range(self.num_wires):
+            #    tensor = np.reshape(self.zero_state, [1, 2, 1])  # this shape is required, even for end nodes it seems
+            #    node = self._add_node(tensor, wires=[w], name="ZeroState")
+            #    nodes.append(node)
+            #    if w > 0:
+            #        tn.connect(nodes[w-1][2], nodes[w][0])
+            ## Note: might want to set canonicalize=False
+            # self.mps = tn.matrixproductstates.finite_mps.FiniteMPS(nodes)
+
+    def _add_initial_state_nodes(self, tensors, wires, names):
+        """Create the nodes representing the initial input state circuit.
+        
+           Input states can be factorized or entangled. If a state can be factorized
+           into :math:`k` subsystems, then ``tensors``, ``wires``, and ``names`` should be lists of length :math:`k`.
```

Comment:
Adjustment to reflect the docstring
```suggestion
           into :math:`k` subsystems, then ``tensors``, ``wires``, and ``names`` should be sequences of length :math:`k`.
```